opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/InterfaceToNodeCacheDaoImpl.java

```diff
@@ -307,15 +307,33 @@ private void dataSourceSyncWithinTransaction() {
      * @return The node ID of the IP Address if known.
      */
     @Override
-    public synchronized Iterable<Integer> getNodeId(final String location, final InetAddress address) {
+    public synchronized List<Integer> getNodeId(final String location, final InetAddress address) {
         if (address == null) {
-            return Collections.emptySet();
+            return Collections.emptyList();
         }
 
         m_lock.readLock().lock();
         try {
-            return Iterables.transform(m_managedAddresses.get(new Key(location, address)),
-                    Value::getNodeId);
+            final var values = m_managedAddresses.get(new Key(location, address));
+            if (values.isEmpty()) {
+                return Collections.emptyList();
+            } else {
+                return values.stream().map(Value::getNodeId).collect(Collectors.toList());
+            }
+        } finally {
+            m_lock.readLock().unlock();
+        }
+    }
+
+    @Override
+    public synchronized Optional<Integer> getFirstNodeId(String location, InetAddress ipAddr) {
+        if (ipAddr == null) {
+            return Optional.empty();
+        }
+        m_lock.readLock().lock();
+        try {
+            var s = m_managedAddresses.get(new Key(location, ipAddr));
```

Comment:
Please use some more descriptive name here instead of `s`, e.g. `values`

Reply:
done